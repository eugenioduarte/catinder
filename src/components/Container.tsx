import { useTheme } from '@/hooks/useTheme'
import { Theme } from '@/types/theme.type'
import React from 'react'
import { StyleSheet, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

interface ContainerProps {
  children: React.ReactNode
}

const Container = ({ children }: ContainerProps) => {
  const theme = useTheme()
  const styles = containerStyles(theme)

  return (
    <View style={styles.container}>
      <SafeAreaView style={styles.safeArea}>{children}</SafeAreaView>
    </View>
  )
}

export default Container

const containerStyles = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
    },
    safeArea: {
      flex: 1,
    },
  })
