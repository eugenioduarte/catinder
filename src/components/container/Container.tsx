import { useTheme } from '@/hooks/useTheme'
import React from 'react'
import { GestureHandlerRootView } from 'react-native-gesture-handler'
import { SafeAreaView } from 'react-native-safe-area-context'
import { containerStyles } from './Container.styles'

interface ContainerProps {
  children: React.ReactNode
}

const Container = ({ children }: ContainerProps) => {
  const theme = useTheme()
  const styles = containerStyles(theme)

  return (
    <GestureHandlerRootView style={styles.container}>
      <SafeAreaView style={styles.safeArea}>{children}</SafeAreaView>
    </GestureHandlerRootView>
  )
}

export default Container
