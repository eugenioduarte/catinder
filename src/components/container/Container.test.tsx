import { render } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'
import Container from './Container'

jest.mock('react-native-gesture-handler', () => {
  return {
    GestureHandlerRootView: ({ children }: any) => <>{children}</>,
  }
})

jest.mock('react-native-safe-area-context', () => {
  return {
    SafeAreaView: ({ children }: any) => <>{children}</>,
  }
})

jest.mock('@/hooks/useTheme', () => ({
  useTheme: () => ({ primaryColor: 'blue' }),
}))

jest.mock('./Container.styles', () => ({
  containerStyles: (theme: any) => ({
    container: { backgroundColor: theme.primaryColor },
    safeArea: { flex: 1 },
  }),
}))

describe('Container Component', () => {
  it('renders children', () => {
    const { getByText } = render(
      <Container>
        <Text>Test Child</Text>
      </Container>,
    )
    expect(getByText('Test Child')).toBeTruthy()
  })
})
