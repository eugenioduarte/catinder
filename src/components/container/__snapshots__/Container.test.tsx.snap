// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Container Component matches snapshot 1`] = `
<Text>
  Snapshot Child
</Text>
`;
